### ISSUE TITLE
GLUE results not reproducible

### ISSUE THREAD

**Issue by lumliolum** [IntelLabs/academic-budget-bert#18]
Hello,

I understand the results mentioned in paper for GLUE are for test set but we are not able to reproduce them. Our pretrained model had a loss of `1.72` and after sweeping through the hyper-parameters mentioned in Table 7 of paper, the best score that we got on CoLA is `43%` (in validation set) and as per table 4 your test result is `57.1%`. We did finetuning on 1 GPU.

Are we missing something ?

**Comment by chutaklee** [comment]
Looks like your model isn't fully trained. I ditched time based training and trained for 23k steps and achieved val loss ~1.6. Test result on CoLA is 56.5.

**Comment by lumliolum** [comment]
Hello @chutaklee 

If you don't mind can you please share the following details of your training. It will help us to try to replicate.

1. number of gpus and gpu specifications.
2. Train batch size ( `train_batch_size` argument for pretraining.py).
3. Micro batch size per gpu and Gradient accumulation steps.

Thanks.

**Comment by chutaklee** [comment]
all here https://wandb.ai/ctl/budget-bert-pretraining/runs/1pno2mq2/overview?workspace=user-ctl

**Comment by peteriz** [comment]
Hi @lumliolum, we provided the hyperparamters we used for sweeping on GLUE tasks.
As @chutaklee posted, the best 24hBERT was trained for ~22-23k steps of bsz 4k (so 95M +/-  sequences), so as far as you manage to replicate a similar (time-based schedule) training session, you should get a model very close to what we got.
Let me know if you have additional questions.

Peter

**Comment by leoozy** [comment]
@lumliolum Could you share your error on proprocessing. As we just preporcess the data using the given scripts

**Comment by leoozy** [comment]
> Looks like your model isn't fully trained. I ditched time based training and trained for 23k steps and achieved val loss ~1.6. Test result on CoLA is 56.5.



> Looks like your model isn't fully trained. I ditched time based training and trained for 23k steps and achieved val loss ~1.6. Test result on CoLA is 56.5.

But the test results reported in paper for CoLA is 57.1. Do you have any idea about the reported results?

**Comment by chutaklee** [comment]
@leoozy 

It's just one percent improvement and honestly I don't bother. Could be hyperparams for finetuning, could be random seeds.

**Comment by leoozy** [comment]
> @leoozy
> 
> It's just one percent improvement and honestly I don't bother. Could be hyperparams for finetuning, could be random seeds.

Hello, when I am training the bert, I noticed that the grad for the first few epochs is Nan. I am not sure it is right for training a Bert. And this brings instability to the bert pretraining process. For example, you may always get Nan grad even you just change the arch of Bert a little

**Comment by leoozy** [comment]
> @leoozy
> 
> It's just one percent improvement and honestly I don't bother. Could be hyperparams for finetuning, could be random seeds.

Hello, can you reporduce other glue resutls in other dataset? I trained bert-large for 23k steps following the give hyparameters. But the acc on eval is only 55%.
